frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -300,7 +303,9 @@ const commonTemplatesUpdater = ({ id, prevState, dispatch, getState }: UpdateOpt
   const [data, isBase64] = CloudInitDataHelper.getUserData(
     toShallowJS(iGetCommonTemplateCloudInit(iTemplate))?.cloudInitNoCloud,
   );
-  const typeData = CloudInitDataHelper.toCloudInitNoCloudSource(data, isBase64);
+  // generate password
+  const updatedData = data?.replace(/(?<=password:\s+).*/, generateCloudInitPassword());
+  const typeData = CloudInitDataHelper.toCloudInitNoCloudSource(updatedData, isBase64);
   const hasCloudInitData = !!typeData?.userData;
 
```

Comment:
this will remove custom cloudinit storage on cloudinit change even if the templates do not include it. Also the type should not be VMWizardStorageType.UI_INPUT.